app/core/src/main/resources/static/js/pages/edit-table-of-contents.js

```diff
@@ -613,6 +618,49 @@ document.addEventListener("DOMContentLoaded", function () {
     updateEmptyStateButton();
   }
 
+  // Undo button funcionality
+  const undoBtn = document.getElementById("undoBtn");
+
+  function saveBookmarkState() {
+    previousBookmarks = [...bookmarks];
+    undoBtn.disabled = false;
+  }
+
+  function undoBookmarkChanges() {
+    if (previousBookmarks === null) return;
+
+    bookmarks = [...previousBookmarks];
+    previousBookmarks = null;
+    undoBtn.disabled = true;
+    updateBookmarksUI();
+  }
+  undoBtn.addEventListener("click", undoBookmarkChanges);
+
+  // ctrl+z keybind
+  function keyPressHandler(event) {
+    // do not override normal paste behavior on input fields
+    if (event.target.tagName.toLowerCase() === "input") return;
+
+    if (event.ctrlKey && event.keyCode == 90) {
```

Comment:
please change to
if ((event.ctrlKey || event.metaKey) && (event.key === 'z' || event.key === 'Z')) {